src/bundle.c

```diff
@@ -29,15 +32,32 @@ static gboolean mksquashfs(const gchar *bundlename, const gchar *contentdir, GEr
 		goto out;
 	}
 
-	sproc = r_subprocess_new(G_SUBPROCESS_FLAGS_STDOUT_SILENCE,
-			&ierror, "mksquashfs",
-			contentdir,
-			bundlename,
-			"-all-root",
-			"-noappend",
-			"-no-progress",
-			"-no-xattrs",
-			NULL);
+	g_ptr_array_add(args, g_strdup("mksquashfs"));
+	g_ptr_array_add(args, g_strdup(contentdir));
+	g_ptr_array_add(args, g_strdup(bundlename));
+	g_ptr_array_add(args, g_strdup("-all-root"));
+	g_ptr_array_add(args, g_strdup("-noappend"));
+	g_ptr_array_add(args, g_strdup("-no-progress"));
+	g_ptr_array_add(args, g_strdup("-no-xattrs"));
+
+
+	if (r_context()->mksquashfs_args != NULL) {
+		res = g_shell_parse_argv(r_context()->mksquashfs_args, &mksquashfs_argcp, &mksquashfs_argvp, NULL);
```

Comment:
You need to pass `ierror` as the last argument to `g_shell_parse_argv()` if you later on want to propagate potential errors upwards.

Reply:
Ah, my glib ignorance again. Interestingly I tested the error case locally and it seemed to propagate just fine, guessing it would have given a bit more context though.